Automattic/jetpack
# Widgets: enqueue stylesheet directly in the widget display method

Fixes #39813

## Proposed changes:

Instead of relying on a conditional check to see if the widget is active, let's call the widgets' enqueuing methods directly from within the widget display method. That should simplify things, and ensure compatiblity with systems that are not compatible with `is_active_widget`.

### Other information:


## Jetpack product discussion

* No

## Does this pull request change what data or activity we track or use?

* No

## Testing instructions:

1. This is easier to test on a site using a classic theme with multiple widget areas. Twenty Ten is a good choice. Install it.
2. Edit the theme's `sidebar.php` and `sidebar-footer.php` files, and add conditionals around some of the widget areas. For example you could change `if ( is_active_sidebar( 'secondary-widget-area' ) ) :` into `if ( is_active_sidebar( 'secondary-widget-area' ) && is_single() ) :` in `sidebar.php`, so the secondary sidebar is only visible when looking at single posts.
3. Install and activate the Classic Widgets plugin.
4. Connect Jetpack to WordPress.com.
5. Purchase a complete plan (you'll need a plan to be able to add the Pay with PayPal widget)
6. Under Jetpack > Settings > Writing, activate the extra sidebar widgets at the bottom of the page.
7. Go to Appearance > Widgets
8. Add widgets to different sidebar areas, loaded on different pages.
9. Configure those widgets so they display information.
10. View the site's frontend, and view source. Search for the different stylesheets that should be loaded for each widget.
    * Those stylesheets should only be loaded when the widget is present on the page.
11. Now install and activate the Elementor plugin.
12. Go to Pages > Add New
13. Enter the Elementor editor.
14. Add a new container.
15. In that container, add the Top Posts widget (scroll down to the "WordPress" part of the inserter, you should see the widget there)
16. Publish the page.
17. View it on the frontend.
    * The widget's stylesheet should be loaded on the page.
